sdk/endpoint/composite/connection.go

```diff
@@ -87,6 +88,14 @@ func (cce *ConnectionCompositeEndpoint) Close(ctx context.Context, connection *c
 	return &empty.Empty{}, nil
 }
 
+func (cce *ConnectionCompositeEndpoint) generateIfName() string {
+	ifName := "nsm" + cce.id.MustGenerate()
+	ifName = strings.Replace(ifName, "-", "", -1)
```

Comment:
Why are we doing this twice?

Reply:
Right, I am not even sure we need to replace these characters. Was assuming it may cause issues with vpp-agent and/or vpp itself if the name contains `-` or `_`. Felt safer this way.